ntbs-service-unit-tests/Services/AlertServiceTest.cs

```diff
@@ -104,5 +105,88 @@ public async Task CreateAlertsForUnmatchedLabResults_CallsAddAlertRangeWithExpec
             Assert.Equal(1, alert.NotificationId);
             Assert.Equal("1", alert.SpecimenId);
         }
+
+        [Fact]
+        public async Task DismissAllOpenAlerts_DismissesOpenAlerts()
+        {
+            // Arrange
+            const int notificationId = 1;
+            var alert1 = new DataQualityTreatmentOutcome12 { AlertId = 101, AlertStatus = AlertStatus.Open };
+            var alert2 = new DataQualityClinicalDatesAlert { AlertId = 102, AlertStatus = AlertStatus.Open };
+            var alerts = new List<Alert> { alert1, alert2 };
+
+            _mockAlertRepository
+                .Setup(r => r.GetAllOpenAlertsByNotificationId(notificationId))
+                .Returns(Task.FromResult(alerts));
+
+            // Act
+            await _alertService.DismissAllOpenAlerts(notificationId);
+
+            // Assert
+            AssertAlertClosedRecently(alert1);
+            AssertAlertClosedRecently(alert2);
+            _mockAlertRepository.Verify(r => r.SaveAlertChangesAsync(NotificationAuditType.Edited), Times.Once);
+        }
+
+        [Theory]
+        [InlineData(NotificationStatus.Closed, true)]
+        [InlineData(NotificationStatus.Denotified, false)]
+        [InlineData(NotificationStatus.Draft, false)]
+        [InlineData(NotificationStatus.Deleted, false)]

```

Comment:
Why don't we want to dismiss the alert if the duplicate is deleted?

Reply:
@JoshCadney I'm going make most of the review tomorrow but thought I would reply to this because this is a bit of a bigger-picture question.

I think we do want to dismiss all alerts when a notification is denotified or deleted, as well as when it is closed. However, this has not been implemented, and was not in the original scope of this ticket, but is closely related. If we do want this feature for delete/dismiss as well (maybe we can chat at standup to decide?) then I'm up for re-doing bits of this ticket where it checks notification status, to check for more than just the notification being closed.

Reply:
I've done this now, so alerts are closed during denotification and deletion as well as during closing notifications.